base/MAPL_EASEConversion.F90

```diff
@@ -0,0 +1,828 @@
+  ! ============================
+  !
+  ! easeV1_conv.F90 - Fortran routines for conversion of azimuthal 
+  !                   equal area and equal area cylindrical grid coordinates
+  ! 
+  ! 30-Jan-1992 H.Maybee
+  ! 20-Mar-1992 Ken Knowles  303-492-0644  knowles@kryos.colorado.edu
+  ! 16-Dec-1993 MJ Brodzik   303-492-8263  brodzik@jokull.colorado.edu
+  !              Copied from nsmconv.f, changed resolutions from 
+  !              40-20-10 km to 25-12.5 km
+  ! 21-Dec-1993 MJ Brodzik   303-492-8263  brodzik@jokull.colorado.edu
+  !              Fixed sign of Southern latitudes in ease_inverse.
+  ! 12-Sep-1994 David Hoogstrate 303-492-4116 hoogstra@jokull.colorado.edu
+  ! 	       Changed grid cell size. Changed "c","f" to "l","h"
+  ! 25-Oct-1994 David Hoogstrate 303-492-4116 hoogstra@jokull.colorado.edu
+  ! 	       Changed row size from 587 to 586 for Mercator projection
+  ! 11-May-2011 reichle: Changed "smap" to "easeV1".  
+  !                      Added SSM/I and AMSR-E "M25" grid.
+  !                      So far ONLY for cylindrical grids.
+  !                      Converted from *.f to *.F90 module
+  ! 
+  ! $Log$
+  ! Revision 1.1  2011-05-11 21:58:46  rreichle
+  ! Adding utilities to map between EASE grids and lat/lon coordinates.
+  !
+  ! Revision 1.3  1994/11/01 23:40:43  brodzik
+  ! Replaced all references to 'ease' with 'smap'
+  ! Replaced all references to 'smap' with 'easeV1' -- reichle
+  !
+  !
+  ! CHANGELOG (easeV2_conv.F90):
+  ! ============================
+  !
+  ! easeV2_conv.F90 - FORTRAN routines for converting grid coordinates
+  !                   (latitude/longitude <--> row/column indices)
+  !                   of the Equal Area Scalable Earth, version 2 (EASEv2) grid
+  !
+  !    ***** ONLY cylindrical ('M') projection implemented *****
+  !
+  ! Ported from Steven Chan's matlab code (smapease2inverse.m, 
+  ! smapease2forward.m), which has been ported from NSIDC's IDL code
+  ! (wgs84_convert.pro, wgs84_inverse.pro) available from  
+  ! ftp://sidads.colorado.edu/pub/tools/easegrid/geolocation_tools/
+  !
+  ! Official references:
+  !  doi:10.3390/ijgi1010032
+  !  doi:10.3390/ijgi3031154 -- correction of M25 "map_scale_m" parameters!
+  !
+  ! 04-Apr-2013 - reichle
+  ! 11-Sep-2018 - reichle, mgirotto -- added 'M25' grid parameters 
+  !
+  !
+  ! CHANGELOG (EASE_conv.F90):
+  ! ==========================
+  !
+  ! 2022-09-13, wjiang+reichle: 
+  !   merged easeV1_conv.F90 and easeV2_conv.F90 into EASE_conv.F90
+  !   - using different values for PI in easeV1 and easeV2 calcs as in old easeV[x]_conv.F90 modules; 
+  !       in contrast, LDAS_EASE_conv.F90 in GEOSldas used only a single value for PI.
+  !   - bug fix in easeV2_get_params() for EASEv2/M25 (to compute s0, divide by 2.0 not by integer 2)
+  !
+  !
+  ! ==========================================================================
+  use, intrinsic :: iso_fortran_env, only: REAL64
+  use MAPL_Constants, only: PI => MAPL_PI_r8
```

Comment:
We have to keep the v1- and v2-specific PI values.  Otherwise, we will lose backward compatibility for creating bcs, and we would also have lat/lons in published SMAP L4 products that aren't consistent with SMAP L1-3 products published by other institutions.